simSPI/linear_simulator/shift_utils.py

```diff
@@ -0,0 +1,90 @@
+"""Module containing method to shift an image in Fourier domain."""
+import numpy as np
+import torch
+
+
+class Shift(torch.nn.Module):
+    """class containing method to shift an image in Fourier domain.
+
+    Written by H. Gupta and Y.S.G. Nashed.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the shift
+
+    """
+
+    def __init__(self, config):
+        super(Shift, self).__init__()
+
+        self.config = config
+        self.frequency = 1.0 / (self.config.side_len * self.config.pixel_size)
+
+        n2 = float(self.config.side_len // 2)
+        ax = torch.arange(-n2, n2 + self.config.side_len % 2)
+        ax = torch.flip(ax, dims=[0])
+        mx, my = torch.meshgrid(ax, ax)
+
+        self.register_buffer("mx", mx.clone())
+        self.register_buffer("my", my.clone())
+
+    def phase_shift(self, x_fourier, t_x, t_y):
+        """Output modulated (fourier equivalent of shifting in primal domain).
+
+        Input images given in batch wise format.
+        The modulation depends on t_x, t_y.
+
+        Parameters
+        ----------
+        x_fourier : torch.Tensor
+            batch of input images (Bx1xSizexSize) in Fourier domain
+        t_x: torch.Tensor
+            batch of shifts (B,) along horizontal axis in Angstrom (A)
+        t_y: torch.Tensor
+            batch of shifts (B,) along vertical axis in Angstrom (A)
+
+        Returns
+        -------
+        output: torch.Tensor
+            batch of modulated fourier images (Bx1xSizexSize) given by
+            output(f_1,f_2)=e^{-2*pi*j*[f_1,f_2]*[t_x, t_y] }*input(f_1,f_2)
+        """
+        t_y = t_y[:, None, None, None]
+        t_x = t_x[:, None, None, None]
+
+        modulation = torch.exp(
+            -2 * np.pi * 1j * self.frequency * (self.mx * t_y + self.my * t_x)
+        )
+
+        return x_fourier * modulation
+
+    def forward(self, x_fourier, shift_params={}):
```

Comment:
Why would it not be possible to do what [Deepsource suggests to do](https://deepsource.io/gh/compSPI/simSPI/run/e9234511-4b7)?

My understanding is that it would look something like this:
```python
    def forward(self, x_fourier, shift_params=None):
        if shift_params is not None:
           x_fourier = self.phase_shift(
               x_fourier, shift_params["shift_x"], shift_params["shift_y"]
            )
        return x_fourier
```